client/http-client/src/client.rs

```diff
@@ -77,10 +79,17 @@ impl HttpClientBuilder {
 		self
 	}
 
+	///
+	pub fn custom_headers(mut self, headers: HeaderMap) -> Self {
```

Comment:
I would rather have a more unified API for the WS client and HTTP client, so if you could add a similar API as https://github.com/paritytech/jsonrpsee/blob/master/client/ws-client/src/lib.rs#L125-#L128 I would be happy with it.

The reason why is to avoid to require users to use `http::HeaderMap`

Reply:
How about now?

Reply:
sorry, I think it would make more sense to return an error instead of `unwrap` these might fail for example if `HeaderName` is uppercase and similar.

in addition an already existing header will just be overwritten with `append`.

so might be better what you did initially for simplicity and change the WS client API instead.

Reply:
//cc @jsdw any opinions?

Reply:
I Added a custom error variant, and made the method fallible. I think this is the nicest we can make this without exposing extraneous APIs.

I also would like to keep the ability to overwrite default headers, what if user wants to send `Content-Type: application/json+jsonrpc` instead of the default `Content-type: application/json`

Reply:
makes sense,

I think the docs should explain that you can overwrite the default headers too perhaps the `doc tests/examples` regarding that would awesome :)